zerver/views/user_groups.py

```diff
@@ -35,6 +35,10 @@ def edit_user_group(request: HttpRequest, user_profile: UserProfile,
         return json_error(_("No new data supplied"))
 
     user_group = access_user_group_by_id(user_group_id, realm=user_profile.realm)
+    group_member_ids = get_members_user_group(user_group)
+
+    if (user_profile.id not in group_member_ids and not user_profile.is_realm_admin):
```

Comment:
Check for user_profile.is_realm_admin first as it is (1) and the other is (N)

Fix other occurances as well